src/Symfony/Component/Routing/Router.php

```diff
@@ -187,7 +188,19 @@ public function getOption($key)
     public function getRouteCollection()
     {
         if (null === $this->collection) {
-            $this->collection = $this->loader->load($this->resource, $this->options['resource_type']);
+            if (null === $this->options['cache_dir'] || null === $this->options['route_collection_cache_file']) {
+                $this->collection = $this->loadRouteCollection();
+            } else {
+                $cache = $this->getConfigCacheFactory()->cache($this->options['cache_dir'].'/'.$this->options['route_collection_cache_file'].'.cache',
+                    function (ConfigCacheInterface $cache) {
+                        $collection = $this->loadRouteCollection();
+
+                        $cache->write(serialize($collection), $collection->getResources());
```

Comment:
var_export is fine for scalar and arrays, but much more complex to use for objects. And this is dealing with an object graph